File: app/src/main/java/xyz/projectplay/jisho/models/Attribution.java
package xyz.projectplay.jisho.models;

class Attribution {
    private boolean jmdict;
    private boolean jmnedict;
    private boolean dppedia;

    Attribution() {

    }

    public boolean isJmdict() {
        return jmdict;
    }

    public boolean isJmnedict() {
        return jmnedict;
    }

    public boolean isDppedia() {
        return dppedia;
    }
}

File: android/src/main/java/com/github/reline/jisho/models/Attribution.kt
/*
 * Copyright 2020 Nathaniel Reline
 *
 * This work is licensed under the Creative Commons Attribution-ShareAlike 4.0 International License.
 * To view a copy of this license, visit http://creativecommons.org/licenses/by-sa/4.0/ or
 * send a letter to Creative Commons, PO Box 1866, Mountain View, CA 94042, USA.
 */

package com.github.reline.jisho.models

import com.github.reline.jisho.network.qualifiers.Dbpedia
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
class Attribution(
    @field:Json(name = "jmdict")
    val isJmdict: Boolean = false,
    @field:Json(name = "jmnedict")
    val isJmnedict: Boolean = false,
    @field:Dbpedia
    val dbpedia: String = false.toString()
) {
    val isDbpedia: Boolean
        get() = dbpedia != false.toString()
}
